istioctl/cmd/version.go

```diff
@@ -29,39 +29,12 @@ import (
 
 	"istio.io/istio/istioctl/pkg/clioptions"
 	"istio.io/istio/istioctl/pkg/multixds"
-	"istio.io/istio/operator/cmd/mesh"
 	"istio.io/istio/pilot/pkg/model"
 	"istio.io/istio/pilot/pkg/xds"
 	"istio.io/istio/pkg/proxy"
 	istioVersion "istio.io/pkg/version"
 )
 
-func newVersionCommand() *cobra.Command {
-	profileCmd := mesh.ProfileCmd()
-	var opts clioptions.ControlPlaneOptions
-	versionCmd := istioVersion.CobraCommandWithOptions(istioVersion.CobraOptions{
-		GetRemoteVersion: getRemoteInfoWrapper(&profileCmd, &opts),
-		GetProxyVersions: getProxyInfoWrapper(&opts),
-	})
-	opts.AttachControlPlaneFlags(versionCmd)
-
-	versionCmd.Flags().VisitAll(func(flag *pflag.Flag) {
-		if flag.Name == "short" {
-			err := flag.Value.Set("true")
-			if err != nil {
-				fmt.Fprintf(os.Stdout, "set flag %q as true failed due to error %v", flag.Name, err)
-			}
-		}
-		if flag.Name == "remote" {
-			err := flag.Value.Set("true")
-			if err != nil {
-				fmt.Fprintf(os.Stdout, "set flag %q as true failed due to error %v", flag.Name, err)
-			}
-		}
-	})
-	return versionCmd
-}
-
 func getRemoteInfo(opts clioptions.ControlPlaneOptions) (*istioVersion.MeshInfo, error) {
```

Comment:
Can you also remove this function if it is not useful? Also I think following funcs are not required
`getRemoteInfoWrapper`
`getProxyInfoWrapper`

Reply:
I think linter caught it https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/27387/lint_istio/19992#1:build-log.txt%3A13

Reply:
fixed!